src/clip.sol

```diff
@@ -314,12 +316,12 @@ contract Clipper {
     }
 
     // --- Shutdown ---
-    function setBreaker(uint256 level) external auth {
+    function setBreaker(uint256 level) external note auth {
```

Comment:
It's worth considering whether triggering the circuit breaker merits its own custom event--in principle `note` is enough, but IIRC custom events are usually a little friendlier to integrators, and the breaker getting tripped would be a pretty important event to be able to reliably detect.

Reply:
Custom events are friendlier in a couple of ways:
1. Can query the log directly rather than needing to query all `LogNotes` and filtering all the with the desired `msg.sig`
2. Has the privilege of logging information that was computed within the function

To my surprise, however, it looks like pulling historical logs or listening to an active event is easier than I expected. [Here's an example in web3.py of setting up an event filter](https://web3py.readthedocs.io/en/stable/filters.html#event-log-filters), which applies a second filter in the `topics` field. If it's as easy as it looks, then my first point is invalid..

Reply:
Should we make a custom event anyways? Or would `note` be enough for integrators?

Reply:
I'm leaning towards `note` being good enough...for comparison, we don't have custom events for `rely`/`deny`/`file`, which are all fairly sensitive and important to monitor. `setBreaker` also lacks any internal values that we might otherwise want to expose.

Reply:
+1 on the `note` being good enough for `setBreaker`